tests/test_views.py

```diff
@@ -557,6 +561,106 @@ def test_detail_view_staff_all_users(self, mock_update):
 
+    @XBlock.register_temp_plugin(StubCourse, 'course')
+    @XBlock.register_temp_plugin(StubSequential, 'sequential')
+    @XBlock.register_temp_plugin(StubHTML, 'html')
+    @patch.object(StubCompat, 'get_cohorts_for_course')
+    def test_stat_view_course_no_cohorts(self, get_cohorts_for_course_mock):
+        get_cohorts_for_course_mock.return_value = None
+        response = self.client.get(self.get_course_stat_url(
+            'edX/toy/2012_Fall',
+            cohorts=1,
+            exclude_roles='staff',
+        ))
+
+        self.assertEqual(response.status_code, 404)
+
+    @XBlock.register_temp_plugin(StubCourse, 'course')
+    @XBlock.register_temp_plugin(StubSequential, 'sequential')
+    @XBlock.register_temp_plugin(StubHTML, 'html')
+    @patch('completion_aggregator.api.v1.views.user_has_excluded_roles')
+    def test_stat_view_staff_user_excluded_from_results(self, excluded_roles_mock):
+        excluded_roles_mock.side_effect = [False, False]
+        self.create_enrollment(user=self.staff_user, course_id=self.course_key)
+        models.Aggregator.objects.submit_completion(
+            user=self.staff_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(
+                block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=4.0,
+            possible=8.0,
+            last_modified=timezone.now(),
+        )
+        response = self.client.get(self.get_course_stat_url(
+            'edX/toy/2012_Fall',
+            cohorts=1,
+            exclude_roles='staff'
+        ))
+
+        self.assertEqual(response.data['results'][0]['completion']['earned'], 1.0)
+        self.assertEqual(response.data['results'][0]['completion']['possible'], 8.0)
+
+    @XBlock.register_temp_plugin(StubCourse, 'course')
+    @XBlock.register_temp_plugin(StubSequential, 'sequential')
+    @XBlock.register_temp_plugin(StubHTML, 'html')
+    @patch('completion_aggregator.api.v1.views.user_has_excluded_roles')
+    def test_stat_view_exclude_user_based_on_role(self, excluded_roles_mock):
+        excluded_roles_mock.side_effect = [False, True]
+        beta_user = User.objects.create(username='beta_user')
+        self.create_enrollment(user=beta_user, course_id=self.course_key)
+
+        models.Aggregator.objects.submit_completion(
+            user=beta_user,
+            course_key=self.course_key,
+            block_key=self.course_key.make_usage_key(
+                block_type='course', block_id='course'),
+            aggregation_name='course',
+            earned=7.0,
+            possible=8.0,
+            last_modified=timezone.now(),
+        )
+
+        response = self.client.get(self.get_course_stat_url(
+            'edX/toy/2012_Fall',
+            cohorts=1,
+            exclude_roles='beta'
+        ))
+        self.assertEqual(excluded_roles_mock.call_count, 2)
```

Comment:
Is this getting called twice because there are two users enrolled in the course?  that's going to be a performance bottleneck in large courses.  We want to make sure we're doing a constant number of queries regardless of the course size.